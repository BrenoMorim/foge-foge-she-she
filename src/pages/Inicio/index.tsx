import Chamada from "components/Chamada";
import MenuEscolherFase from "components/MenuEscolherFase";

export default function Inicio() {
  return (
    <section>
      <MenuEscolherFase />
      <Chamada />
    </section>
  );
}