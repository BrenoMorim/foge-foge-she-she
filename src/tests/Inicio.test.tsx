import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Inicio from '../pages/Inicio';
import fases_disponiveis from 'data/fases';

const mockNavegacao = jest.fn();

jest.mock('react-router-dom', () => {
    return {
        useNavigate: () => mockNavegacao
    }
});

test('A página inicial deve ser renderizada corretamente', () => {
    const paginaInicio = render(<Inicio/>);
    expect(paginaInicio).toMatchSnapshot();
});

describe('A página inicial', () => {

    beforeEach(() => {
        render(<Inicio/>);
    });

    test('Deve conter o botão de iniciar jogo desabilitado se não houver fase selecionada', () => {
        const botaoJogar = screen.getByText("Jogar");
        expect(botaoJogar).toBeDisabled();
    });

    test('Deve conter select com as fases disponíveis', () => {
        const numeroDeFases = fases_disponiveis.length;
        const selecionarFase = screen.getByLabelText("Escolha uma fase:");
        expect(selecionarFase).toBeRequired();

        const fasesDisponiveis = screen.getAllByRole("option");

        // Já tem uma option por padrão
        expect(fasesDisponiveis).toHaveLength(numeroDeFases + 1); 
    });

    test('Deve conter a seção chamada', () => {
        const sectionChamada = screen.getByTestId('chamada');
        expect(sectionChamada).toBeInTheDocument();
    });

    test('Deve conseguir selecionar uma fase e então jogar', () => {
        const selecionarFase = screen.getByLabelText("Escolha uma fase:");
        fireEvent.change(selecionarFase, {
            target: {
                value: fases_disponiveis[0].id
            }
        });
        
        const botao = screen.getByText('Jogar');
        expect(botao).toBeEnabled();
        fireEvent.click(botao);
        expect(mockNavegacao).toBeCalledTimes(1);
    });
});