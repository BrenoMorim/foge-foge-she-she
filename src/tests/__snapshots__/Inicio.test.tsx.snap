// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A página inicial deve ser renderizada corretamente 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <form
          class="menu-fases base__painel-roxo base__flex-container"
        >
          <label
            class="menu-fases__rotulo"
            for="fases"
          >
            Escolha uma fase:
          </label>
          <select
            class="menu-fases__fases base__painel-branco"
            id="fases"
            name="fases"
            required=""
          >
            <option
              disabled=""
              selected=""
              value="1"
            >
              Fases disponíveis:
            </option>
            <option
              value="1"
            >
              Tutorial
            </option>
            <option
              value="2"
            >
              Meio fácil
            </option>
            <option
              value="3"
            >
              Mais ou menos
            </option>
            <option
              value="4"
            >
              Nível médio
            </option>
            <option
              value="5"
            >
              Pouquinho difícil
            </option>
            <option
              value="6"
            >
              Sorrisinho =)
            </option>
            <option
              value="7"
            >
              Coelhinho
            </option>
            <option
              value="8"
            >
              Divisão
            </option>
            <option
              value="9"
            >
              Labirinto
            </option>
            <option
              value="10"
            >
              Colado
            </option>
            <option
              value="11"
            >
              Emparedado
            </option>
            <option
              value="12"
            >
              Espiral
            </option>
            <option
              value="13"
            >
              Pilha
            </option>
            <option
              value="14"
            >
              Rosquinha
            </option>
          </select>
          <button
            class="base__botao"
            disabled=""
            role="button"
            type="submit"
          >
            Jogar
          </button>
        </form>
        <section
          class="chamada base__flex-container"
          data-testid="chamada"
        >
          <h3
            class="chamada__titulo"
          >
            Ajude o Sherlock a fugir do breno e da jaminha ninjas!
          </h3>
          <div
            class="chamada__imagens"
          >
            <img
              alt="Breno ninja"
              class="chamada__imagens__personagem"
              src="assets/mapa/breno.png"
            />
            <img
              alt="Jaminha ninja"
              class="chamada__imagens__personagem"
              src="assets/mapa/jaminha.png"
            />
          </div>
          <h3
            class="chamada__titulo"
          >
            Ele precisa comer todas as rações ou espantar todos os ninjas com suas garras!
          </h3>
        </section>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <form
        class="menu-fases base__painel-roxo base__flex-container"
      >
        <label
          class="menu-fases__rotulo"
          for="fases"
        >
          Escolha uma fase:
        </label>
        <select
          class="menu-fases__fases base__painel-branco"
          id="fases"
          name="fases"
          required=""
        >
          <option
            disabled=""
            selected=""
            value="1"
          >
            Fases disponíveis:
          </option>
          <option
            value="1"
          >
            Tutorial
          </option>
          <option
            value="2"
          >
            Meio fácil
          </option>
          <option
            value="3"
          >
            Mais ou menos
          </option>
          <option
            value="4"
          >
            Nível médio
          </option>
          <option
            value="5"
          >
            Pouquinho difícil
          </option>
          <option
            value="6"
          >
            Sorrisinho =)
          </option>
          <option
            value="7"
          >
            Coelhinho
          </option>
          <option
            value="8"
          >
            Divisão
          </option>
          <option
            value="9"
          >
            Labirinto
          </option>
          <option
            value="10"
          >
            Colado
          </option>
          <option
            value="11"
          >
            Emparedado
          </option>
          <option
            value="12"
          >
            Espiral
          </option>
          <option
            value="13"
          >
            Pilha
          </option>
          <option
            value="14"
          >
            Rosquinha
          </option>
        </select>
        <button
          class="base__botao"
          disabled=""
          role="button"
          type="submit"
        >
          Jogar
        </button>
      </form>
      <section
        class="chamada base__flex-container"
        data-testid="chamada"
      >
        <h3
          class="chamada__titulo"
        >
          Ajude o Sherlock a fugir do breno e da jaminha ninjas!
        </h3>
        <div
          class="chamada__imagens"
        >
          <img
            alt="Breno ninja"
            class="chamada__imagens__personagem"
            src="assets/mapa/breno.png"
          />
          <img
            alt="Jaminha ninja"
            class="chamada__imagens__personagem"
            src="assets/mapa/jaminha.png"
          />
        </div>
        <h3
          class="chamada__titulo"
        >
          Ele precisa comer todas as rações ou espantar todos os ninjas com suas garras!
        </h3>
      </section>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
